app/client/components/liveChat/liveChat.tsx

```diff
@@ -254,16 +336,47 @@ export const WithLiveChatContainer = () => {
       window.sessionStorage.getItem("liveChat") === "1" &&
       liveChatContainerRef.current
     ) {
-      initLiveChat(liveChatContainerRef.current);
+      if (window.guardian && window.guardian.identityDetails.userId) {
+        initLiveChat(
+          liveChatContainerRef.current,
+          window.guardian.identityDetails.userId
+        );
+      } else {
+        initLiveChat(liveChatContainerRef.current, "");
+      }
     }
   }, []);
 
-  return <div ref={liveChatContainerRef} css={withLiveChatContainerCss} />;
+  return <div ref={liveChatContainerRef} css={liveChatCss} />;
 };
 
 export const StartLiveChatButton = () => {
-  function bootstrapChat(): void {
-    window.embedded_svc.bootstrapEmbeddedService();
+  function updatePreChatEmailField(): void {
+    const preChatEmailField = document.getElementById(
+      "SuppliedEmail"
+    ) as HTMLInputElement;
+    if (window.guardian.identityDetails.email && preChatEmailField) {
+      // tslint:disable-next-line:no-object-mutation
+      preChatEmailField.value = window.guardian.identityDetails.email;
```

Comment:
Could we remove the need for the event if we use this line of code instead:

```
            embedded_svc.settings.prepopulatedPrechatFields = {SuppliedEmail: window.guardian.identityDetails.email}; 

```

Reply:
Correct me if I'm wrong but if we opt for this code and the user is not logged in then the  `embedded_svc.settings.prepopulatedPrechatFields.SuppliedEmail` would override the email entered in the pre chat text box and end up passing along an empty string?